example/ndpiReader.c

```diff
@@ -874,81 +885,84 @@ void printCSVHeader() {
   fprintf(csv_fp, "\n");
 }
 
-static int cache_idx_from_name(const char *name)
-{
-  if(strcmp(name, "ookla") == 0)
-    return NDPI_LRUCACHE_OOKLA;
-  if(strcmp(name, "bittorrent") == 0)
-    return NDPI_LRUCACHE_BITTORRENT;
-  if(strcmp(name, "zoom") == 0)
-    return NDPI_LRUCACHE_ZOOM;
-  if(strcmp(name, "stun") == 0)
-    return NDPI_LRUCACHE_STUN;
-  if(strcmp(name, "tls_cert") == 0)
-    return NDPI_LRUCACHE_TLS_CERT;
-  if(strcmp(name, "mining") == 0)
-    return NDPI_LRUCACHE_MINING;
-  if(strcmp(name, "msteams") == 0)
-    return NDPI_LRUCACHE_MSTEAMS;
-  if(strcmp(name, "stun_zoom") == 0)
-    return NDPI_LRUCACHE_STUN_ZOOM;
-  return -1;
-}
-
-static int parse_cache_param(char *param, int *cache_idx, int *param_value)
+static int parse_two_unsigned_integer(char *param, u_int32_t *num1, u_int32_t *num2)
 {
-  char *saveptr, *tmp_str, *cache_str, *param_str;
-  int idx;
+  char *saveptr, *tmp_str, *num1_str, *num2_str;
 
   tmp_str = ndpi_strdup(param);
   if(tmp_str) {
-    cache_str = strtok_r(tmp_str, ":", &saveptr);
-    if(cache_str) {
-      param_str = strtok_r(NULL, ":", &saveptr);
-      if(param_str) {
-        idx = cache_idx_from_name(cache_str);
-        if(idx >= 0) {
-          *cache_idx = idx;
-          *param_value = atoi(param_str);
-          ndpi_free(tmp_str);
-	  return 0;
-	}
+    num1_str = strtok_r(tmp_str, ":", &saveptr);
+    if(num1_str) {
+      num2_str = strtok_r(NULL, ":", &saveptr);
+      if(num2_str) {
+        *num1 = atoi(num1_str);
+        *num2 = atoi(num2_str);
+        ndpi_free(tmp_str);
+	return 0;
       }
     }
   }
   ndpi_free(tmp_str);
   return -1;
 }
 
-static int parse_two_unsigned_integer(char *param, u_int32_t *num1, u_int32_t *num2)
+static int parse_three_strings(char *param, char **s1, char **s2, char **s3)
 {
-  char *saveptr, *tmp_str, *num1_str, *num2_str;
+  char *saveptr, *tmp_str, *s1_str, *s2_str = NULL, *s3_str;
 
   tmp_str = ndpi_strdup(param);
   if(tmp_str) {
-    num1_str = strtok_r(tmp_str, ":", &saveptr);
-    if(num1_str) {
-      num2_str = strtok_r(NULL, ":", &saveptr);
-      if(num2_str) {
-        *num1 = atoi(num1_str);
-        *num2 = atoi(num2_str);
+    if(param[0] == ',') { /* First parameter might be missing */
+      s1_str = NULL;
+      s2_str = strtok_r(tmp_str, ",", &saveptr);
+    } else {
+      s1_str = strtok_r(tmp_str, ",", &saveptr);
+      if(s1_str) {
+        s2_str = strtok_r(NULL, ",", &saveptr);
+      }
+    }
+    if(s2_str) {
+      s3_str = strtok_r(NULL, ",", &saveptr);
+      if(s3_str) {
+        *s1 = ndpi_strdup(s1_str);
+        *s2 = ndpi_strdup(s2_str);
+        *s3 = ndpi_strdup(s3_str);
         ndpi_free(tmp_str);
-	return 0;
+        if(!s1 || !s2 || !s3) {
+          ndpi_free(s1);
+          ndpi_free(s2);
+          ndpi_free(s3);
+          return -1;
+        }
+        return 0;
       }
     }
   }
   ndpi_free(tmp_str);
   return -1;
 }
 
+int __add_cfg(char *proto, char *param, char *value, int dup)
```

Comment:
What about:

```suggestion
int ndpi_add_cfg(char *proto, char *param, char *value, int dup)
```

or 

```suggestion
int reader_add_cfg(char *proto, char *param, char *value, int dup)
```

Reply:
Done. I choose the second option. I prefer to keep the "ndpi_" prefix for the public API of the library